src/qibosoq/programs/sweepers.py

```diff
@@ -46,24 +39,20 @@ def add_sweep_info(self, sweeper: Sweeper):
         Args:
             sweeper (RfsocSweep): single qibolab sweeper object to register
         """
-        starts = sweeper.starts
-        stops = sweeper.stops
-
         sweep_list = []
         sweeper.parameters = [Parameter(par) for par in sweeper.parameters]
-        sweeper.starts = np.array(sweeper.starts)
-        sweeper.stops = np.array(sweeper.stops)
+
         if sweeper.parameters[0] is Parameter.BIAS:
             for idx, jdx in enumerate(sweeper.indexes):
                 gen_ch = self.qubits[jdx].dac
                 sweep_type = "gain"
                 std_register = self.get_gen_reg(gen_ch, sweep_type)
                 swept_register = self.new_gen_reg(gen_ch, name=f"sweep_bias_{gen_ch}")
-                self.bias_sweep_registers[gen_ch] = (swept_register, std_register)
+                self.bias_sweep_registers[gen_ch] = (swept_register, std_register)  # type: ignore
 
                 max_gain = int(self.soccfg["gens"][gen_ch]["maxv"])
-                starts = (sweeper.starts * max_gain).astype(int)
-                stops = (sweeper.stops * max_gain).astype(int)
+                starts = (np.array(sweeper.starts) * max_gain).astype(int)
```

Comment:
Which is the type of `sweeper.starts`?

Reply:
it's a list

Reply:
Why not an `np.array` in the first place? Is there a case in which you want to `.append()` or something like that?

Reply:
It was because the communication protocol is "based" on json and ndarray are not natively serializable... Clearly I can convert them to list before communication. The two options seem pretty equivalent to me

Reply:
Ok, you're right.

I believe we should just pick a scheme: converting immediately, or as late as possible.

This of course is only possible when data have a "true format" and a different representation. In which case, I would argue that the *as soon as possible* scheme is preferable, since you already know you're going to consume data in that fashion, and you'd avoid doing it multiple times.
The advantage of doing it late would be to never do it when it's not strictly needed (but I believe this is more rarely happening).

So, I would directly deserialize into an array, when you load the JSON representation into the object (if possible).